File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <cctype>

using namespace std;

string EngConvert(string str){
    
    if(str == "zero"){
        return "0";
    }
    else if(str == "one"){
        return "1";
    }
    else if(str == "two"){
        return "2";
    }
    else if(str == "three"){
        return "3";
    }
    else if(str == "four"){
        return "4";
    }
    else if(str == "five"){
        return "5";
    }
    else if(str == "six"){
        return "6";
    }
    else if(str == "seven"){
        return "7";
    }
    else if(str == "eight"){
        return "8";
    }
    else if(str == "nine"){
        return "9";
    }
    
    return "";    
}

int solution(string s) {
    int answer = 0;
    
    string str(""), tmp("");
    for(char c : s){
        if(isalpha(c)){
            tmp += c;
            string eng = EngConvert(tmp);
            if( eng == "")
                continue;
            else{
                str += eng; 
                tmp.clear();
            }
            
        }else{
            str += c;
        }
    }
    
    for(char c : str){
        answer = (10 * answer + (c - 48));
    }
    
    return answer;
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
import java.util.concurrent.ConcurrentHashMap;

class Solution {
   private int str2int(String str){
        ConcurrentHashMap<String, Integer> hashMap = new ConcurrentHashMap<>();
        hashMap.put("zero", 0 );
        hashMap.put("one", 1);
        hashMap.put("two", 2);
        hashMap.put("three", 3);
        hashMap.put("four", 4);
        hashMap.put("five", 5);
        hashMap.put("six", 6);
        hashMap.put("seven", 7);
        hashMap.put("eight",8);
        hashMap.put("nine", 9);

        //get value
        Integer num = hashMap.get(str);
        if(num == null){
            return -1;
        }

        return num.intValue();


    }
    public int solution(String s) {



        int answer = 0;
        StringBuilder builder = new StringBuilder();
        StringBuilder tmp = new StringBuilder();
        for(char c : s.toCharArray()){
            if(!Character.isAlphabetic(c)){
                builder.append(c);
            }
            else{
                tmp.append(c);
                int result = str2int(tmp.toString());
                if(result != -1){
                    builder.append(result);
                    tmp.setLength(0);
                }

            }
        }
        answer = Integer.parseInt(builder.toString());

        return answer;
    }

}